query-planner-js/src/buildPlan.ts

```diff
@@ -1185,13 +1200,28 @@ class FetchGroup {
       outputRewrites: outputRewrites.length === 0 ? undefined : outputRewrites,
     };
 
+    const selectionSetNode = this.selection.toSelectionSetNode()
+    const selections = selectionSetNode.selections
+    const selectionNode = selections[0]
+    const conditions = collectConditionalValues(selectionNode)
+
+    if (conditions.size > 0) {
+      return this.isTopLevel
+          ? computeConditionNodeForTopLevelIncludeAndSkip(fetchNode, conditions)
+          : {
+            kind: 'Flatten',
+            path: this.mergeAt!,
+            node: computeConditionNodeForTopLevelIncludeAndSkip(fetchNode, conditions),
```

Comment:
Ideally this will never return an undefined node from here then figure out a way to typecast this as conditionNode itself.

Reply:
But For now the return type of this function is (PlanNode | undefined) which is problematic for us